src/vmm/src/pipe.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 Sartura All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+#![cfg(target_arch = "x86_64")]
+
+use std::io::{Read, Write};
+
+use std::fs::{File};
+use std::os::unix::io::{AsRawFd, RawFd};
+
+use nix::sys::stat;
+use nix::unistd;
```

Comment:
I think `nix` is a pretty big (new) dependency to pull in just for named pipes. `libc` is already a dependency and a basic safe wrapper over `libc::mkfifo` could be added to the `utils` crate:
```rust
pub enum Error {
    Io(io::Error),
    Path,
}

pub fn mkfifo<P: AsRef<Path>>(fifo_path: P, mode: u32) -> result::Result<(), Error> {
    let fifo_path_str = fifo_path.as_ref().to_str().ok_or(Error::Path)?;
    let ret = unsafe { libc::mkfifo(fifo_path_str.as_ptr() as *const i8, mode) };
    if ret != 0 {
        Err(Error::Io(io::Error::last_os_error()))
    } else {
        Ok(())
    }
}
```
(This is just a suggestion and can be simplified a lot by just working with a `&str` instead of generics.)